detector/other/FieldMapBrBz.cpp

```diff
@@ -52,71 +72,54 @@ void FieldMapBrBz::fieldComponents(const double* pos , double* globalField) {
   const double posPhi = atan2(pos[1], pos[0]);
 
   //Mokka defines these things as x and y and phi
-  double x = posRho;
-  double y = posZ;
+  double r = posRho;
+  double z = posZ;
   double phi = posPhi;
 
   //Get positive values to do less checks when comparing
-  if( y < 0 ) {
-    y = -y ;
-    phi += M_PI ;
+  if( z < 0 ) {
+    z   *= -1;
+    phi += M_PI;
   }
 
+  if(r < rhoMin) r = rhoMin;
+  if(z < zMin  ) z = zMin;
+
   //APS: Note the mokka field map does not start at 0, so we have to assume that
   //this area is covered, or add some more parameters for the values where the
   //field is supposed to be. Now we just assume it starts at 0/0/0, outside there is no field
-  if (not ( 0.0 <= x && x <= rhoMax &&
-	    0.0 <= y && y <= zMax ) ) {
+  if (not ( rhoMin <= r && r <= rhoMax &&
```

Comment:
Ok, right. But then you can also drop the comparison with rhoMin/zMin, because that is now always true.
And please change the comment above and move it to before setting r to rhoMin etc.